sdk/provisioning/Azure.Provisioning/src/Primitives/BicepValueReference.cs

```diff
@@ -25,7 +25,7 @@ internal BicepExpression GetReference(bool throwIfNoRoot = true)
             }
             else if (throwIfNoRoot)
             {
-                throw new NotImplementedException("Cannot reference a construct without a name.");
+                throw new InvalidOperationException("Cannot reference a construct without a name.");
```

Comment:
Is this a breaking change?  If so I think its good because not implemented is the wrong exception but we should mention in the changelog which public APIs have this behavioral change.

Reply:
In public API, there is no breaking change.
But this would be a behavioral breaking change. It would break customer's code if they wrote explicitly the exception.